### ISSUE TITLE
Proxied Vaadin routes are rebuilt instead of reused

### ISSUE THREAD

**Issue by jcgueriaud1** [vaadin/flow#16062]
### Description of the bug

When my application has this spring annotation:
```
@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true, jsr250Enabled = true)
```

The route is wrapped as a Spring proxy.
![image](https://user-images.githubusercontent.com/51313578/221593683-ff0446cb-e0e9-45b3-89df-d468523db2f2.png)

If I navigate from the route to the same route then the route is rebuilt instead of being reused.
![image](https://user-images.githubusercontent.com/51313578/221593922-0c6107a9-4570-4d56-9e87-f5d23e4523c1.png)

### Expected behavior

The route should be reused and not rebuilt.

AbstractNavigationStateRenderer#getRouteTarget should return the component instead of creating a new one.

### Minimal reproducible example

https://github.com/jcgueriaud1/spring-proxy-issue

Go on the helloworld page and navigate with the routerlink or the button.

### Versions

- Vaadin / Flow version: 23
- Java version: 11 or 17

**Comment by knoobie** [comment]
Isn't this by design? Or am I missing something? If you don't define a scope for the View / Route it's "request scope / once only used" and therefore navigating again to the view should create a new instance? Or is it the special case where the User moves from View1 to View1 instead of View1 -> View2 -> View1? Which often doesn't even trigger a navigation if I remember correctly

**Comment by jcgueriaud1** [comment]
I can't answer if it's by design or not.
But without Spring, if you 'navigate' to the same view the component is found in the `ui.getInternals().getActiveRouterTargetsChain()`
https://github.com/vaadin/flow/blob/ee00826e610873b750569abcf70541e11d0c4010/flow-server/src/main/java/com/vaadin/flow/router/internal/AbstractNavigationStateRenderer.java#L126

With Spring it's also found.

If you're adding the method security, the class in the activerouter target chain is a Spring proxy so it's not found.

You could do something like this:
`ui.getInternals().getActiveRouterTargetsChain().get(0).getElement().getComponent().get().getClass()`

To get the inner class but it's a bit ugly

**Comment by mcollovati** [comment]
This issue has been resolved in #19151, for Vaadin 24, but not back-ported to Vaadin 23
